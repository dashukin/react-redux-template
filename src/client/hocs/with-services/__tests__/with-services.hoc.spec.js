import React from 'react';
import { shallow, mount } from 'enzyme';
import {
  ServicesProvider, withServices,
} from '../with-services.hoc';

describe('with-services.hoc', () => {
  let mockServices;

  beforeEach(() => {
    mockServices = {};
  });

  describe('ServicesProvider', () => {
    it('should match snapshot', () => {
      const component = (
        <ServicesProvider services={mockServices}>Children</ServicesProvider>
      );
      const wrapper = shallow(component);

      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('withServices', () => {
    it('should pass services to child component', () => {
      const ChildComponent = () => 'Child component';
      const WrappedComponent = withServices(ChildComponent);

      const component = (
        <ServicesProvider services={mockServices}>
          <WrappedComponent />
        </ServicesProvider>
      );

      const wrapper = mount(component);

      expect(wrapper.find(ChildComponent).prop('services')).toBe(mockServices);
    });
  });
});
