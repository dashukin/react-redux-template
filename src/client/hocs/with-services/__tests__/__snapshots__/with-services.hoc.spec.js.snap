// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`with-services.hoc ServicesProvider should match snapshot 1`] = `
<ContextProvider
  value={Object {}}
>
  Children
</ContextProvider>
`;
